asteroidsyncservice/watch.h

```diff
@@ -45,6 +52,10 @@ class Watch : public QObject
     QString weatherCityName();
     quint8 batteryLevel();
     bool timeServiceReady();
+    bool screenshotServiceReady();
+    unsigned int screenshotProgress();
+    QString screenshotPath() const;
+    QString screenshotFilename() const;
```

Comment:
What's the rationale behind having two separate properties for screenshotPath and for screenshotFilename?
Also, could you make screenshotFilename (or its combination with screenshotPath if you end up changing them) a format string that can be parsed by https://doc.qt.io/qt-5/qdatetime.html#toString everytime a new screenshot is saved? so that the QML code doesn't have to care about time handling and changing screenshotFilename all the time, they could just give a format string once and for all and avoid conflicts

Reply:
At the moment i did this on the qml side
`root.watch.screenshotPath = "/home/phablet/.local/share/telescope.asteroidos/screenshot/"
                    root.watch.screenshotFilename = "screenshot" + Qt.formatDateTime(new Date(), "ddMMyyyy_hhmmss") + ".jpg"`

Reply:
If you do this on the QML side then this property is only interpreted once at startup, and then never again, so the date will actually never be updated.

Reply:
This gets updated on every screenshot trigger
`onClicked: {
                    root.watch.requestScreenshot()
                    root.watch.screenshotPath = "/home/phablet/.local/share/telescope.asteroidos/screenshot/"
                    root.watch.screenshotFilename = "screenshot" + Qt.formatDateTime(new Date(), "ddMMyyyy_hhmmss") + ".jpg"
                    pageStack.push("qrc:/qml/pages/ScreenshotPage.qml", {watch: watch})
                }`

If i have the path separate,  then i can easily check if this directory exists or need to create that.

Reply:
I would argue that just seting screenshotPath and screenshotFilename once and for all would be a much nicer interface. This is how declarative programming is supposed to work. in QML you should ideally never have to write iterative code in javascript, as much as possible should be done by setting properties to what you want and then let the code do what is right when necessary.

I would prefer if the property could be set once with something like:
screenshotPath: "/home/phablet/.local/share/telescope.asteroidos/screenshot/"
screenshotFilename: "screenshotddMMyyyy_hhmmss.jpg"

The API that is exposed to users shouldn't be affected by implementation details. You can easily extract the base directory of a given URL in C++ using https://doc.qt.io/archives/qt-4.8/qfileinfo.html#dir and then QML doesn't have to do those things

Reply:
I've been tried implementing your ideas and suggestions. I'm looking forward to your review :)